package/MDAnalysis/analysis/density.py

```diff
@@ -497,12 +495,53 @@ def __repr__(self):
             grid_type = 'histogram'
         return '<Density ' + grid_type + ' with ' + str(self.grid.shape) + ' bins>'
 
+def _set_user_grid(gridcenter, xdim, ydim, zdim, smin, smax):
+    """Helper function to set the grid dimensions to user defined values
+       
+       Called by : density_from_Universe 
+       Condition : gridcenter must be defined
+
+       Parameters
+       ----------
+       gridcenter : numpy array containing the x,y,z of the box center (np.float32)
+       x/y/zdim :   box edge lengths in each dimension
+       smin/smax :  minimum and maximum x,y,z for given selection (padding ignored)
+
+       Returns
+       -------
+       umin/umax :  replacement values for the box min and max values (np.float32)
+       
+    """
+    #Copy grid to umin and umax and then set the min/max to half the edge values
+    umin = np.copy(gridcenter)
+    umax = np.copy(gridcenter)
+
+    #set x dimension
+    umin[0] -= xdim/2.0
+    umax[0] += xdim/2.0
+    #set y dimension
+    umin[1] -= ydim/2.0
+    umax[1] += ydim/2.0
+    #set z dimension
+    umin[2] -= zdim/2.0
+    umax[2] += zdim/2.0
+
+    # Here we test if coords of selection fall outside of the defined grid
+    # if this happens, we warn users they may want to resize their grids
+    if any(smin < umin) or any(smax > umax):
+        msg = "Atom selection does not fit grid --- you may want to define a larger box"
+        logger.warning(msg)
```

Comment:
Wouldn't it be better to raise an exception here?

Reply:
So my initial idea of going for a warning here, is down to the fact that I still wanted to allow for use cases where one could have a selection that is larger than the box of interest and have the code only histogram the atoms that fall within the grid. The example use case here could be to have a selection of all the waters within a given radius of a ligand that is slightly larger than the box of interest, thus ensuring that the grid accounts for all the waters within it.

Unfortunately I couldn't think of a way to set an exception for this that would yield a clean warning to users so I went for the logger. That being said I am not well versed in exception handling either, I was wondering if there was a specific idea that you had in mind?

Note: in 33b66a7 I do add an assert at the top of the _set_user_defined_grid function to check that the  gridcenter is a 3 element ndarray.